packages/xo-server/src/xo-mixins/subjects.js

```diff
@@ -14,7 +14,7 @@ import { forEach, isEmpty, lightSet, mapToArray } from '../utils'
 const log = createLogger('xo:xo-mixins:subjects')
 
 const addToArraySet = (set, value) =>
-  set && !includes(set, value) ? set.concat(value) : [value]
+  set ? (includes(set, value) ? set : set.concat(value)) : [value]
```

Comment:
You can explicit the test and use the native method `Array.prototype.includes`.

Suggestion: 
```js
set !== undefined ? (set.includes(value) ? set : set.push(value)) : [value]
```

Note: `set` will never be undefined because `user.groups` and `group.user` are always defined. See: 
https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/models/user.js#L55
https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-server/src/models/group.js#L38

Reply:
1. explicit test :+1: 
2. native `includes` :+1: 
3. `addToArraySet` and `removeFromArraySet` accept `undefined` sets, let's not change that
4. using `push` is a behavior change!!!

Reply:
Also, regarding 3, I believe it was not always the case.